axlearn/experiments/text/gpt/fuji.py

```diff
@@ -419,6 +423,57 @@ def get_trainer_kwargs(
                 ),
             ),
         )
+    elif model_size == "405B":
+        remat_policy = config_for_function(
+            jax_remat_policies.save_and_offload_only_these_names
+        ).set(
+            names_which_can_be_saved=[],
+            names_which_can_be_offloaded=[
+                "FlashAttention.q_proj",
+                "FlashAttention.k_proj",
+                "FlashAttention.v_proj",
+                "TransformerLayer.input",
+            ],
+            offload_src="device",
+            offload_dst="pinned_host",
+        )
+        trainer_kwargs = dict(
+            model_kwargs=dict(
+                num_layers=126,
+                hidden_dim=53248,
+                num_heads=128,
+                num_kv_heads=8,
+                rope_theta=rope_theta,
+                flash_attention=flash_attention,
+            ),
+            learner_kwargs=dict(peak_lr=8e-5, weight_decay=0.1),
+            max_sequence_length=max_sequence_length,
+            train_batch_size=train_batch_size,
+            max_step=max_step,
+            mesh_shape=mesh_shape_from_axes(fsdp=-1),
+            mesh_rules=(
+                # tpu-v6e.
+                (
+                    "tpu-v6e-.*",
+                    ChainConfigModifier.default_config().set(
+                        config_modifiers=[
+                            MeshShapeModifier.default_config().set(
+                                mesh_shape=mesh_shape_from_axes(data=-1, fsdp=256)
```

Comment:
note this requires optimizer state weight only offloading PR to be merged: https://github.com/apple/axlearn/pull/789